pkg/synthetictests/static_pod.go

```diff
@@ -109,7 +110,7 @@ func testStaticPodLifecycleFailure(events monitorapi.Intervals, kubeClientConfig
 			matches := regexp.MustCompile("to ([0-9]+) because static pod is ready").FindStringSubmatch(event.Note)
 			if len(matches) == 2 {
 				reachedRevision, _ := strconv.ParseInt(matches[1], 0, 64)
-				if isRevisionUpdate && isForNode && reachedRevision == staticPodFailure.revision {
+				if isRevisionUpdate && isForNode && reachedRevision >= staticPodFailure.revision {
```

Comment:
`>` isn't appropriate here since we want to reach the desired revision, not a future level.